lib/Model/Url.php

```diff
@@ -52,39 +55,25 @@ public static function fromUrl(string $url): self
 
     public function __toString(): string
     {
-        return rtrim($this->uri->__toString(), '/');
+        return (string) $this->uri;
     }
 
     public function resolveUrl(string $link, ReferringElement $referringElement = null): self
     {
         try {
-            $link = Uri::createFromString($link);
-        } catch (Exception $e) {
+            $parsedLink = HttpUri::createFromString($link);
+        } catch (\Throwable $e) {
             throw new InvalidUrl($e->getMessage(), 0, $e);
         }
 
-        if ($link->getPath()) {
-            $link = $link->withPath($this->normalizePath($link));
-        }
-
-        if (!$link->getScheme()) {
-            $link = $link->withScheme($this->uri->getScheme());
-        }
-
-        if (!$link->getHost()) {
-            $link = $link->withHost($this->uri->getHost());
-        }
+        $resolvedLink = FollowRedirects::resolve($this->uri, $parsedLink);
 
-        if (!$link->getPort()) {
-            $link = $link->withPort($this->uri->getPort());
-        }
-
-        if ($link->getFragment()) {
+        if ($resolvedLink->getFragment()) {
```

Comment:
This condition should be removed, otherwise a fragment of `0` is kept.

Reply:
Updated this to `'' !==`